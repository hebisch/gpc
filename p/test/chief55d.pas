program two;
type
THandle = integer;

function f: integer;
begin
  f := 99
end;

function LoadLibrary (const s: String): THandle;
begin
  Writeln (s);
  LoadLibrary := 42
end;

function GetProcAddress (h: THandle; const s: String): Pointer;
begin
  Writeln (h, ' ', s);
  GetProcAddress := @f
end;

function FreeLibrary (h: THandle): Boolean;
begin
  Writeln (h);
  FreeLibrary := h > 0
end;

type
pf = function : integer;

var
h : THandle = LoadLibrary ('winstl32.dll');
v : pf = pf (GetProcAddress (h, 'Chief32DllVersion'));

begin
  Writeln ('begin');
  Writeln ('Handle=', h);
  Writeln ('V is assigned=', Assigned (v)); { TRUE }
  Writeln ('Version=', v); { wrong result }
  Writeln ('DLL Unloaded=',FreeLibrary (h)); { TRUE }
end.
