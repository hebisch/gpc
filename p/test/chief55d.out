winstl32.dll
42 Chief32DllVersion
begin
Handle=42
V is assigned=True
Version=99
DLL Unloaded=42
True
